src/runtime/PythonTypes/PyInt.cs

```diff
@@ -230,5 +230,7 @@ public string ToString(string format, IFormatProvider formatProvider)
             using var _ = Py.GIL();
             return ToBigInteger().ToString(format, formatProvider);
         }
+
+        public override TypeCode GetTypeCode() => TypeCode.Int64;
```

Comment:
Generally, this is wrong.

Reply:
The docs are a bit unclear on this function. I used this as an "should almost always work" value, the `InvalidCastException` might still be thrown if the conversion fails. We could drop the `GetTypeCode` overrides entirely and just return `TypeCode.Object` if you'd prefer that.

Reply:
I can't say for others, just for this one. Honestly, not sure if anyone ever uses this.

Reply:
I'm not sure I follow. `GetTypeCode` is supposed to be a hint that allows you to write a `switch` like
```csharp
switch (convertible.GetTypeCode()) {
    case TypeCode.Int64:
        return Convert.ToInt64(convertible);
}
```
It doesn't mean that the object *is* a `System.Int64` already.

Reply:
You seem to be correct on that. Still not sure `Int64` is valid for this specific case, but float and string probably are. I am fine either way though. I don't think we will get people complaining one way or another any time soon.